backend/graphql/resolvers/testSessionResolvers.ts

```diff
@@ -30,75 +29,62 @@ const classService: IClassService = new ClassService(
-              `Student id ${result.student} not found in class ${testSession.class.id}`,
-            );
-          }
-          return {
-            ...result,
-            student: studentDTO,
-          };
-        }),
-      };
-    },
-    testSessions: async (): Promise<TestSessionResponseDTO[]> => {
-      return testSessionService.getAllTestSessions();
-    },
-    testSessionByAccessCode: async (
+    testSession: (_parent: undefined, { id }: { id: string }) =>
+      testSessionService.getTestSessionById(id),
+    testSessions: () => testSessionService.getAllTestSessions(),
+    testSessionByAccessCode: (
       _parent: undefined,
       { accessCode }: { accessCode: string },
-    ): Promise<TestSessionResponseDTO> => {
-      return testSessionService.getTestSessionByAccessCode(accessCode);
-    },
-    testSessionsByTeacherId: async (
+    ) => testSessionService.getTestSessionByAccessCode(accessCode),
+    testSessionsByTeacherId: (
       _parent: undefined,
       { teacherId }: { teacherId: string },
-    ): Promise<Array<TestSessionResponseDTO>> => {
-      return testSessionService.getTestSessionsByTeacherId(teacherId);
-    },
+    ) => testSessionService.getTestSessionsByTeacherId(teacherId),
   },
   Mutation: {
-    createTestSession: async (
+    createTestSession: (
       _req: undefined,
       { testSession }: { classId: string; testSession: TestSessionRequestDTO },
-    ): Promise<TestSessionResponseDTO> => {
-      return testSessionService.createTestSession(testSession);
-    },
-    createTestSessionResult: async (
+    ) => testSessionService.createTestSession(testSession),
+    createTestSessionResult: (
       _req: undefined,
       { id, result }: { id: string; result: ResultRequestDTO },
-    ): Promise<TestSessionResponseDTO> => {
-      return testSessionService.createTestSessionResult(id, result);
-    },
-    deleteTestSession: async (
-      _req: undefined,
-      { id }: { id: string },
-    ): Promise<string> => {
-      return testSessionService.deleteTestSession(id);
+    ) => testSessionService.createTestSessionResult(id, result),
+    deleteTestSession: (_req: undefined, { id }: { id: string }) =>
+      testSessionService.deleteTestSession(id),
+  },
+  TestSessionResponseDTO: {
+    class: ({ class: parentClass }: TestSessionResponseDTO) =>
+      classService.getClassById(parentClass),
+    teacher: ({ teacher }: TestSessionResponseDTO) =>
+      userService.getUserById(teacher),
+    school: ({ school }: TestSessionResponseDTO) =>
+      schoolService.getSchoolById(school),
+    test: ({ test }: TestSessionResponseDTO) => testService.getTestById(test),
+    results: async ({
+      class: parentClass,
+      results,
+    }: TestSessionResponseDTO) => {
+      if (!results) return [];
+
+      // Returning all students whenever results are requested is not ideal, but
+      // it's better than making the same request for each result when the
+      // students are needed.
```

Comment:
i think it should be fine no? cuz there aren't use cases where we request specific students for now

Reply:
Tru